nessie/src/main/java/org/apache/iceberg/nessie/NessieCatalog.java

```diff
@@ -0,0 +1,431 @@
+          "no message");
+    } catch (NessieNotFoundException e) {
+      throw new RuntimeException("Failed to drop table as ref is no longer valid.", e);
+    } catch (NessieConflictException e) {
+      throw new RuntimeException("Failed to drop table as table state needs to be refreshed.");
+    }
+
+    // TODO: purge should be blocked since nessie will clean through other means.
+    if (purge && lastMetadata != null) {
+      BaseMetastoreCatalog.dropTableData(ops.io(), lastMetadata);
+    }
+    // TODO: fix this so we don't depend on it in tests.
+    refresh();
+    return true;
+  }
+
+  @Override
+  public void renameTable(TableIdentifier from, TableIdentifier toOriginal) {
+    reference.checkMutable();
+
+    TableIdentifier to = removeCatalogName(toOriginal);
+
+    IcebergTable existingFromTable = table(from);
+    if (existingFromTable == null) {
+      throw new NoSuchTableException("table %s doesn't exists", from.name());
+    }
+    IcebergTable existingToTable = table(to);
+    if (existingToTable != null) {
+      throw new AlreadyExistsException("table %s already exists", to.name());
+    }
+
+    Operations contents = ImmutableOperations.builder()
+        .addOperations(ImmutablePut.builder().key(toKey(to)).contents(existingFromTable).build(),
+            ImmutableDelete.builder().key(toKey(from)).build())
+        .build();
+
+    try {
+      client.getTreeApi().commitMultipleOperations(reference.getAsBranch().getName(), reference.getHash(),
+          "iceberg rename table", contents);
+      // TODO: fix this so we don't depend on it in tests.
+      refresh();
+    } catch (Exception e) {
+      throw new CommitFailedException(e, "failed");
+    }
+  }
+
+  private TableIdentifier removeCatalogName(TableIdentifier to) {
+
+    String[] levels = to.namespace().levels();
+    // check if the identifier includes the catalog name and remove it
+    if (levels.length >= 2 && name().equalsIgnoreCase(to.namespace().level(0))) {
+      Namespace trimmedNamespace = Namespace.of(Arrays.copyOfRange(levels, 1, levels.length));
+      return TableIdentifier.of(trimmedNamespace, to.name());
+    }
+
+    // return the original unmodified
+    return to;
+  }
+
+  public TreeApi getTreeApi() {
+    return client.getTreeApi();
+  }
+
+  public void refresh() {
+    reference.refresh();
+  }
+
+  public String getHash() {
+    return reference.getHash();
+  }
+
+  public static Builder builder(Configuration conf) {
+    return new Builder(conf);
+  }
+
+  /**
+   * creating namespaces in nessie is implicit, therefore this is a no-op. Metadata is ignored.
+   *
+   * @param namespace a multi-part namespace
+   * @param metadata a string Map of properties for the given namespace
+   */
+  @Override
+  public void createNamespace(Namespace namespace, Map<String, String> metadata) {
+  }
+
+  @Override
+  public List<Namespace> listNamespaces(Namespace namespace) throws NoSuchNamespaceException {
```

Comment:
Should we create a trait just for listing namespaces that are implicit?

Reply:
How do you mean? A new interface that no-ops create, load, drop from `SupportsNamespaces`?

Reply:
Just an interface that omits those. All this needs is to list namespaces, not do anything else.